### ISSUE TITLE
ARIA improvements for navLinks

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#6524]
A [navLink](https://fullcalendar.io/docs/date-nav-links) is a date-number within a cell that does something when clicked.

It is constructed with an `<a>` tag that has a `tabindex="0"`, which inserts it into the tab order. Unfortunately, the `<a>` tag does not have an `href` attribute, so nothing happens when the user presses SPACE or ENTER when the navLink is focused.

**SOLUTION:** define keypress handlers for ENTER and SPACE that executes the click action.

IN THE FUTURE, when we can make breaking DOM changes, we'll turn this element into a `<button>` tag that looks like a link.

Another issue is that the navLink does not have a useful description for screen readers. We should offer a new option. The locale will supply the default:

```js
{
  navLinkAriaLabel: "Go to $0" // will yield "Go to January 4th, 2018"
}
```

This setting will accept a function as well.
